apps/opik-backend/src/main/java/com/comet/opik/infrastructure/bi/InstallationReportService.java

```diff
@@ -0,0 +1,133 @@
+package com.comet.opik.infrastructure.bi;
+
+import com.comet.opik.infrastructure.OpikConfiguration;
+import com.comet.opik.infrastructure.lock.LockService;
+import com.google.inject.ImplementedBy;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import ru.vyarus.dropwizard.guice.module.lifecycle.GuiceyLifecycle;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+@ImplementedBy(InstallationReportServiceImpl.class)
+interface InstallationReportService {
+    String NOTIFICATION_EVENT_TYPE = "opik_os_startup_be";
+
+    void reportInstallation();
+}
+
+@Singleton
+@Slf4j
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+class InstallationReportServiceImpl implements InstallationReportService {
+
+    private final @NonNull LockService lockService;
+    private final @NonNull UsageReportService usageReport;
+    private final @NonNull OpikConfiguration config;
+    private final @NonNull Client client;
+
+    public void reportInstallation() {
+
+        String eventType = GuiceyLifecycle.ApplicationStarted.name();
+
+        if (!config.getUsageReport().isEnabled()) {
+            log.info("Usage report is disabled");
+            return;
+        }
+
+        if (StringUtils.isEmpty(config.getUsageReport().getUrl())) {
+            log.warn("Usage report URL is not set");
+            return;
+        }
+
+        var lock = new LockService.Lock("opik-%s".formatted(eventType));
+
+        try {
+            lockService.executeWithLock(lock, tryToReportStartupEvent(eventType))
+                    .subscribeOn(Schedulers.boundedElastic())
+                    .block();
```

Comment:
Minor: maybe in this case it's recommended to have a timeout. We don't in resource calls as we have both REST server side and client side timeouts, but this call path is different.

Reply:
Sure, I will add it to the subsequent PR.